From f759a3ed382cc58e701f8d80b64c34e629a13ff2 Mon Sep 17 00:00:00 2001
From: Wright Feng <wright.feng@cypress.com>
Date: Tue, 5 May 2020 08:21:17 -0500
Subject: [PATCH 106/138] Revert "brcmfmac: validate ifp pointer in
 brcmf_txfinalize"

This reverts commit "brcmfmac: validate ifp pointer in brcmf_txfinalize"
because the related change has been in upstream, commit c80d26e81ef1
("brcmfmac: fix WARNING during USB disconnect in case of unempty psq")

Signed-off-by: Wright Feng <wright.feng@cypress.com>

---
 drivers/net/wireless/broadcom/brcm80211/brcmfmac/core.c | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/core.c b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/core.c
index 95381b8..00d95f1 100644
--- a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/core.c
+++ b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/core.c
@@ -552,11 +552,6 @@ void brcmf_txfinalize(struct brcmf_if *ifp, struct sk_buff *txp, bool success)
 	struct ethhdr *eh;
 	u16 type;
 
-	if (!ifp) {
-		brcmu_pkt_buf_free_skb(txp);
-		return;
-	}
-
 	eh = (struct ethhdr *)(txp->data);
 	type = ntohs(eh->h_proto);
 
-- 
1.9.1

